02-Custom-Login/app/controllers/home_controller.rb

```diff
@@ -0,0 +1,35 @@
+require "auth0"
+
+class HomeController < ApplicationController
+  attr_reader :client
+
+  def auth
+    v2_creds = { client_id: ENV['AUTH0_CLIENT_ID'],
+     token: '',
+     api_version: 2,
+     domain: ENV['AUTH0_DOMAIN'] }
+
+    @client = Auth0Client.new(v2_creds)
+
+    begin
+      if params[:signup]
+        signup
+      else
+        login
+      end
+      redirect_to '/dashboard'
+    rescue => ex
+      redirect_to '/', notice: ex.message
+    end
+  end
+
+  private
+
+  def login
+    client.login(params[:user], params[:password], connection:'Username-Password-Authentication')
```

Comment:
The failure is displayed in line 22. It redirects to home with ex.message and displayed in a label